mentat/code_file_manager.py

```diff
@@ -1,52 +1,61 @@
 import logging
 import os
 from pathlib import Path
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any
 
 from termcolor import cprint
 
 from mentat.parsers.file_edit import FileEdit
 
-from .code_file import CodeFile
 from .config_manager import ConfigManager
 from .errors import MentatError
 from .user_input_manager import UserInputManager
+from .utils import sha256
 
 if TYPE_CHECKING:
     # This normally will cause a circular import
     from .code_context import CodeContext
 
 
+def _read_file(config: ConfigManager, path: Path) -> list[str]:
+    abs_path = path if path.is_absolute() else Path(config.git_root / path)
+    with open(abs_path, "r") as f:
+        lines = f.read().split("\n")
+    return lines
+
+
+class FileLineReader:
+    """A temporary helper class to replace the pre-loaded list of lines."""
+    def __init__(self, config: ConfigManager):
+        self.config = config
+
+    def get(self, path: Path, default: Any=None) -> list[str]:
```

Comment:
I'm not quite sure about this; the reason we always read the files and kept the full file lines in our dictionary before we sent it to GPT is because the user might change the file while the model is running, but because our changes rely on line numbers it won't necessarily work on the new files (which is why we warn the user that the file changed, and if they agree we overwrite their changes with the old file contents). I think we should keep the file_lines dictionary because we still do need to do that check, and we still need to work on the old file lines if the user agrees to overwrite the possibly changed file.

Reply:
That's right, good catch. It needs to be cached somewhere at the time it's sent to the model, then checked against active when an edit is made. I'll keep this pattern but move it somewhere else (I'm intending `CodeFileManager` to work with multiple `Session`s, and that cache should be attached to a session).